pkg/module/gitsource/git.go

```diff
@@ -28,7 +38,7 @@ func (g GitSource) FetchSource(path, repo, version string) (*ocm.Source, error)
 	access := github.New(repo, "", commit)
 
 	sourceMeta := ocm.SourceMeta{
-		Type: sourceType,
```

Comment:
I think it is invalid. The ocm library doesn't help here - the field is just a `string` without any docs.
But...
1) The `CONSUMER_TYPE` constant has been moved to the identity package. This package has relation to authn/authz code in the ocm library (e.g: repository credentials). Here we're just defining a type of sources repository, it's something else.
2) The closer look at the ocm library shows this:
```
 ✓ .../github.com/open-component-model/ocm $ ag -Q '"github"'
docs/reference/ocm_add_source-configuration.md
787:$ ocm add source-config sources.yaml --name sources --type filesystem --access '{ "type": "gitHub", "repoUrl": "github.com/open-component-model/ocm", "commit": "xyz" }'

cmds/ocm/commands/ocmcmds/sourceconfig/add/cmd.go
47:$ ocm add source-config sources.yaml --name sources --type filesystem --access '{ "type": "gitHub", "repoUrl": "github.com/open-component-model/ocm", "commit": "xyz" }'

pkg/contexts/ocm/accessmethods/github/method.go
34:     Type   = "gitHub"
39:     LegacyType   = "github"

pkg/contexts/credentials/builtin/github/identity/identity.go
17:const CONSUMER_TYPE = "Github"
``` 

There is a command that `"add a source specification to a source config file"`. And in this command usage doc you can see the `"type": "gitHub"` is used. This constant matches the one defined in the `pkg/contexts/ocm/accessmethods/github/method.go`.

Conclusion: I think we should use one of the constants defined in the `pkg/contexts/ocm/accessmethods/github/method.go` file here, not the one from the `identity` package.
What do you think?

Reply:
OK, I think we should create another issue to verify that - it's also used in Lifecycle-Manager. This PR is not about changing this particular constant - probably it's not important anyway. I still think this particular value is incorrect, but it was like that for a long time.
Neither OCM code nor the spec is clear on that topic, so I am not going spend more time on it.